src/service-worker.test.js

```diff
@@ -276,12 +278,69 @@ test('SW should open deep link in click handler if one is provided', () => {
   }
   clickListener(clickEvent);
 
-  // Then the deep link should be opened in a new tab
-  expect(openedWindows).toHaveLength(1);
-  expect(openedWindows[0]).toEqual('https://pusher.com');
+  clickEvent.getWaitUntilPromise().then(() => {
+    // Then the deep link should be opened in a new tab
+    expect(clients).toContainEqual({ url: 'https://pusher.com' });
+
+    // And the notification should be closed
+    expect(clickEvent._isOpen()).toEqual(false);
+  });
```

Comment:
Because the new solution uses `Promise` chaining, I had to implement `getWaitUntilPromise()` for the `clickEvent` like `FakePushEvent`:
https://github.com/pusher/push-notifications-web/blob/ec30388b95ddf763ae17b5b7509f8a99b677f7a0/src/service-worker.test.js#L674-L688

Previously I used `await new Promise(process.nextTick)`, which in Jest will wait until all Promises are completed, but I changed it to this `getWaitUntilPromise()` approach to be more consistent with the existing tests.